weaviate/collection/data.py

```diff
@@ -474,3 +535,30 @@ def reference_add_many(self, from_property: str, refs: List[BatchReference]) ->
             for ref in refs
         ]
         self._reference_add_many(refs_dict)
+
+    @staticmethod
+    def __parse_properties_grpc(data: Dict[str, Any]) -> weaviate_pb2.Properties:
+        multi_target: List[weaviate_pb2.BatchObject.RefPropertiesMultiTarget] = []
+        single_target: List[weaviate_pb2.BatchObject.RefPropertiesSingleTarget] = []
+        non_ref_properties: Struct = Struct()
+        for key, val in data.items():
+            if isinstance(val, ReferenceToMultiTarget):
+                multi_target.append(
+                    weaviate_pb2.BatchObject.RefPropertiesMultiTarget(
+                        uuids=val.uuids, target_collection=val.target_collection, prop_name=key
+                    )
+                )
+            elif isinstance(val, ReferenceTo):
+                single_target.append(
+                    weaviate_pb2.BatchObject.RefPropertiesSingleTarget(
+                        uuids=val.uuids, prop_name=key
+                    )
+                )
+            else:
+                non_ref_properties.update({key: val})
+
+        return weaviate_pb2.BatchObject.Properties(
+            non_ref_properties=non_ref_properties,
+            ref_props_multi=multi_target,
+            ref_props_single=single_target,
+        )
```

Comment:
This function looks identical to that which is in `_DataCollection`. Can they both be abstracted into the base class `_Data` to avoid duplication?

Reply:
I _think_ we need to do something different here, so I seperated it already but haven't tested it yet. As we need to have this in fast, I'd do the model testing later